h2o-genmodel/src/main/java/hex/genmodel/easy/RowToRawDataConverter.java

```diff
@@ -0,0 +1,175 @@
+package hex.genmodel.easy;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.algos.deepwater.DeepwaterMojoModel;
+import hex.genmodel.easy.exception.PredictException;
+import hex.genmodel.easy.exception.PredictNumberFormatException;
+import hex.genmodel.easy.exception.PredictUnknownCategoricalLevelException;
+import hex.genmodel.easy.exception.PredictUnknownTypeException;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+
+/**
+ * This class is intended to transform a RowData instance - for which we want to get prediction to - into a raw array
+ */
+public class RowToRawDataConverter {
+
+  public final GenModel _m;
+  private final HashMap<String, Integer> _modelColumnNameToIndexMap;
+  private final HashMap<Integer, HashMap<String, Integer>> _domainMap;
+  private final EasyPredictModelWrapper.ErrorConsumer _errorConsumer;
+
+  private final boolean _convertUnknownCategoricalLevelsToNa;
+  private final boolean _convertInvalidNumbersToNa;
+  
+  public RowToRawDataConverter(GenModel m,
+                               HashMap<String, Integer> modelColumnNameToIndexMap,
+                               HashMap<Integer, HashMap<String, Integer>> domainMap,
+                               EasyPredictModelWrapper.ErrorConsumer errorConsumer,
+                               boolean convertUnknownCategoricalLevelsToNa,
+                               boolean convertInvalidNumbersToNa) {
+    _m = m;
+    _modelColumnNameToIndexMap = modelColumnNameToIndexMap;
+    _domainMap = domainMap;
+    _errorConsumer = errorConsumer;
+    _convertUnknownCategoricalLevelsToNa = convertUnknownCategoricalLevelsToNa;
+    _convertInvalidNumbersToNa = convertInvalidNumbersToNa;
+  }
+
+  /**
+   * 
+   * @param data instance of RowData we want to get prediction for
+   * @param rawData array that will be filled up from RowData instance and returned
+   * @return `rawData` array with data from RowData. 
+   * @throws PredictException Note: name of the exception feels like out of scope of the class with name `RowToRawDataConverter` 
+   * but this conversion is only needed to make it possible to produce predictions so it makes sense
+   */
+  public double[] convert(RowData data, double[] rawData) throws PredictException {  
+
+    // TODO: refactor
```

Comment:
how about fixing this todo when aready refactoring this code? at least splitting the method into smaller chunks would be most welcome

Reply:
@honzasterba I'm really into refactoring even legacy code which is not really related to a scope of the task. Just in current situation I don't have time for this. I will create a jira for this piece of code for somebody to have fun with it later!